patches/tModLoader/Terraria/Projectile.TML.cs

```diff
@@ -86,6 +88,37 @@ private static void HandlePlayerStatModifiers(IEntitySource spawnSource, Project
 		}
 	}
 
+	/// <summary>
+	/// Will drop loot the same way as when <see cref="ProjectileID.Geode"/> is cracked open.
+	/// </summary>
+	/// <param name="entity">The entity the loot originates from</param>
+	public static void DropGeodeLoot(Entity entity)
+	{
+		var dict = ItemID.Sets.GeodeDrops;
+		foreach (var pair in dict) {
+			string exceptionCommon = $"{Lang.GetItemNameValue(pair.Key)} registered in 'ItemID.Sets.{nameof(ItemID.Sets.GeodeDrops)}'";
+			if (pair.Value.minStack < 1)
+				throw new Exception($"{exceptionCommon} must have minStack bigger than 0");
+
+			if (pair.Value.maxStack <= pair.Value.minStack)
+				throw new Exception($"{exceptionCommon} must have maxStack bigger than minStack");
+		}
+		var list = dict.Keys.ToList();
```

Comment:
Would prefer if this part was done in an `ItemLoader` method, after content is added. Sure people don't open geodes that often, but it doesn't provide a good template to follow. Better to separate initialization, validation, and execution